### ISSUE TITLE
Regression tests are random

### ISSUE THREAD

**Issue by simonduq** [contiki-os/contiki#1630]
Many of the tests are stochastic, with a quite random outcome. This was not very noticeable so far because in case of failure we would re-run the faulty test with a new seed. But since https://github.com/contiki-os/contiki/pull/1622 this is not the case anymore. We have seen PRs that fix code style or other minor aspect result in Travis failure.

I see at least two solutions:
* Revert my change. We go back to re-running with a different seed in case of failure.
* Set the bar higher and fix the tests. We could identify the too random tests by running the current master on e.g. 20 seeds for each test. Only tests that are successful 20 times remain enabled. Others are put on a list of nice goals to achieve (e.g. in a new github issue). Something similar to the results I was showing in this comment: https://github.com/contiki-os/contiki/pull/1403#issuecomment-160740602

Extra suggestion: would be nice to run several seeds in Travis but check that *all* succeed instead of *any*. We cannot afford this for the tests that take a long time, but how about a per-test configuration?

**Comment by joakimeriksson** [comment]
I agree fully. One seed is probably ok - but we might add more precise test coverage by other means (e.g. more tests) rather than more seeds. We should avoid tests that succeeds by luck and tests that fails by bad luck.
